server2/middleware/authentication.js

```diff
@@ -32,7 +32,7 @@ async function authenticateUser(user) {
 }
 
 async function isDuplicateUser(newUser) {
-  const existingUser =  await User.findOne({where: {username: newUser.username}});
+  const existingUser =  await User.findOne({where: {name: newUser.username}});
```

Comment:
it means 2 users can share the same email?

Reply:
there's another isDuplicateEmail function to check duplicate email. Shld i combine the two together?